### ISSUE TITLE
IMPORT FAILED - Can't install custom node

### ISSUE THREAD

**Issue by pneumanon** [AIFSH/ComfyUI-Hallo#21]
I successfully installed this custom node the other day but ran into a number of issues immediately and didn't get it working.  I never resolved all the issues, so I tried updating to the version from today. After trying to update to the latest version of Hallo, though, it simply won't install. I've tried fixing the installation, uninstalling and reinstalling, manually cloning the repo to the custom_nodes folder and no matter what I do, ComfyUI says Hallo is missing. 

(IMPORT FAILED): C:\ComfyUI\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-IP_LAP
(IMPORT FAILED): C:\ComfyUI\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Hallo

**Comment by jackeldestripat** [comment]
Try this in the env:  pip install --upgrade diffusers[torch]
This solve my problem 
This was my error log in the console.
ImportError: cannot import name 'GLIGENTextBoundingboxProjection' from 'diffusers.models.embeddings'

**Comment by pneumanon** [comment]
> Try this in the env: pip install --upgrade diffusers[torch] This solve my problem This was my error log in the console. ImportError: cannot import name 'GLIGENTextBoundingboxProjection' from 'diffusers.models.embeddings'

Thank you for the suggestion. I did run into this error previously and was able to successfully update the embeddings. I no longer have that error, and I did finally get it installed again, but now I'm stuck at a new error.

Unfortunately, trying to get this node working has proven to be extremely challenging and even affected my main ComfyUI installation.  Previously, I had a faceswap workflow that was working just fine, but after trying to install Hallo, that no longer works. The faceswap workflow **does** work on the new ComfyUI Portable installation, but other things don't work on that one, like AnimateDiff. It has an error when running ReActorFaceSwap:

**Could not run 'torchvision::nms'**

Funny thing is, this was one of the first errors I ran into when trying to get Hallo to work when I first had it installed. I solved this and some of the subsequent issues, but more just kept popping up. 

It's hard to even keep track of all the things I've tried, since I've created new instances of ComfyUI Portable and tried so many things. I believe finding the proper xformers installation helped, but that wasn't easy to track down. I'm on torch 2.2.2+cu121 and needed xformers-0.0.25.post1 for it to work. 

The next sticking point that I managed my way through was the audio separator trying to call **Kim_Vocal_2.onnx.onnx** and an easy solution was to just rename the file. But now it's stuck here:

Traceback (most recent call last):
  File "C:\ComfyUI_standalone\ComfyUI\custom_nodes\ComfyUI-Hallo\Hallo\scripts\inference.py", line 379, in <module>
    inference_process(command_line_args)
  File "C:\ComfyUI_standalone\ComfyUI\custom_nodes\ComfyUI-Hallo\Hallo\scripts\inference.py", line 176, in inference_process
    with AudioProcessor(
         ^^^^^^^^^^^^^^^
  File "C:\ComfyUI_standalone\ComfyUI\custom_nodes\ComfyUI-Hallo\Hallo\hallo\datasets\audio_processor.py", line 67, in __init__
    assert self.audio_separator.model_instance is not None, "Fail to load audio separate model."
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Separator' object has no attribute 'model_instance'

**Comment by mdkid79** [comment]
same issues, I run the https://github.com/AIFSH/ComfyUI-Hallo/issues/21#issuecomment-2183099194
but doesn't work, how should I do?
thanks

**Comment by jackeldestripat** [comment]
> > Try this in the env: pip install --upgrade diffusers[torch] This solve my problem This was my error log in the console. ImportError: cannot import name 'GLIGENTextBoundingboxProjection' from 'diffusers.models.embeddings'
> 
> Thank you for the suggestion. I did run into this error previously and was able to successfully update the embeddings. I no longer have that error, and I did finally get it installed again, but now I'm stuck at a new error.
> 
> Unfortunately, trying to get this node working has proven to be extremely challenging and even affected my main ComfyUI installation. Previously, I had a faceswap workflow that was working just fine, but after trying to install Hallo, that no longer works. The faceswap workflow **does** work on the new ComfyUI Portable installation, but other things don't work on that one, like AnimateDiff. It has an error when running ReActorFaceSwap:
> 
> **Could not run 'torchvision::nms'**
> 
> Funny thing is, this was one of the first errors I ran into when trying to get Hallo to work when I first had it installed. I solved this and some of the subsequent issues, but more just kept popping up.
> 
> It's hard to even keep track of all the things I've tried, since I've created new instances of ComfyUI Portable and tried so many things. I believe finding the proper xformers installation helped, but that wasn't easy to track down. I'm on torch 2.2.2+cu121 and needed xformers-0.0.25.post1 for it to work.
> 
> The next sticking point that I managed my way through was the audio separator trying to call **Kim_Vocal_2.onnx.onnx** and an easy solution was to just rename the file. But now it's stuck here:
> 
> Traceback (most recent call last): File "C:\ComfyUI_standalone\ComfyUI\custom_nodes\ComfyUI-Hallo\Hallo\scripts\inference.py", line 379, in inference_process(command_line_args) File "C:\ComfyUI_standalone\ComfyUI\custom_nodes\ComfyUI-Hallo\Hallo\scripts\inference.py", line 176, in inference_process with AudioProcessor( ^^^^^^^^^^^^^^^ File "C:\ComfyUI_standalone\ComfyUI\custom_nodes\ComfyUI-Hallo\Hallo\hallo\datasets\audio_processor.py", line 67, in **init** assert self.audio_separator.model_instance is not None, "Fail to load audio separate model." ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ AttributeError: 'Separator' object has no attribute 'model_instance'


Try this.
pip install audio-separator 

I think that not all libraries are in the requirements.txt

**Comment by pneumanon** [comment]
Unfortunately, nothing I do seems to work. I've created several different instances of ComfyUI-standalone (portable) with no luck.

After reading some other open issues, I though the main problem was that ComfyUI-standalone include python 3.11. So, now I've tried several regular installations of ComfyUI using python 3.10 in a venv. I even tried installing the older version of torch, as exemplified in the installation instructions: torch==2.1.0 torchvision==0.16.0 torchaudio==2.1.0 xformers==0.0.22.post7 --index-url https://download.pytorch.org/whl/cu121

At this point, I've tried torch 2.3.1, 2.3.0, 2.2.2, 2.2.1 and 2.1.0 with matching installations for xformers, torchvision and torchaudio. 

Most of the custom nodes needed to install Hallo fail to install. I get several "No module found" errors when starting ComfyUI. I really want to try this, but I have no idea what else I can try since everything seems to fail and I actually feel like I take several steps back each time I try something new. Any suggestions would be very welcome. I'm willing to try whatever. I have no problem creating new installations of ComfyUI.
